File: app/src/main/java/com/danilomendes/figurines/di/DatabaseModule.java
package com.danilomendes.figurines.di;

import android.app.Application;

import com.danilomendes.figurines.data.local.DatabaseHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by danilo on 08-11-2017.
 */
@Module
public class DatabaseModule {

    @Provides
    @Singleton
    DatabaseHelper provideDatabaseHelper(Application application) {
        return new DatabaseHelper(application);
    }
}

File: app/src/main/java/com/danilomendes/figurines/di/DatabaseModule.kt
package com.danilomendes.figurines.di

import android.app.Application
import com.danilomendes.figurines.data.local.DatabaseHelper
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by danilo on 08-11-2017.
 */
@Module
class DatabaseModule {

    @Provides
    @Singleton
    internal fun provideDatabaseHelper(application: Application): DatabaseHelper {
        return DatabaseHelper(application)
    }
}
